spirv_glsl.hpp

```diff
@@ -64,6 +65,149 @@ typedef uint32_t AccessChainFlags;
+			NV_shader_thread_shuffle,
+			ARB_shader_ballot,
+			ARB_shader_group_vote,
+			AMD_gcn_shader,
+
+			CandidateCount
+		};
+		static std::string getExtensionName(Candidate c)
+		{
+			switch (c)
+			{
+			case NV_gpu_shader_5:
+				return "GL_NV_gpu_shader_5";
+			case NV_shader_thread_group:
+				return "GL_NV_shader_thread_group";
+			case NV_shader_thread_shuffle:
+				return "GL_NV_shader_thread_shuffle";
+			case ARB_shader_ballot:
+				return "GL_ARB_shader_ballot";
+			case ARB_shader_group_vote:
+				return "GL_ARB_shader_group_vote";
+			case AMD_gcn_shader:
+				return "GL_AMD_gcn_shader";
+			default:
+				return "";
+			}
+		}
+		static SmallVector<std::string> getExtraRequiredExtensionNames(Candidate c)
+		{
+			switch (c)
+			{
+			case ARB_shader_ballot:
+				return { "GL_ARB_shader_int64" };
+			case AMD_gcn_shader:
+				return { "GL_AMD_gpu_shader_int64" };
+			default:
+				return {};
+			}
+		}
+
+		enum Feature
+		{
+			SubgroupMask,
+			SubgroupSize,
+			SubgroupInvocationID,
+			SubgroupID,
+			NumSubgroups,
+			SubgroupBrodcast_First,
+			SubgroupBallotFindLSB_MSB,
+			SubgroupAll_Any_AllEqual,
+			SubgroupElect,
+			SubgroupBarrier,
+			SubgroupMemBarrier,
+			SubgroupBallot,
+
+			FeatureCount
+		};
+
+		using Result = std::array<uint32_t, CandidateCount>;
+
+		void requestFeature(Feature ft)
+		{
+			featureMask |= (1u << ft);
+		}
+
+		bool isFeatureRequested(Feature ft) const
+		{
+			return featureMask & (1u << ft);
+		}
+
+		Result resolve() const
+		{
+			Result res;
+
+			for (uint32_t i = 0u; i < FeatureCount; ++i)
+			{
+				if (featureMask & (1u << i))
+				{
+					for (Candidate c : getCandidatesForFeature(static_cast<Feature>(i)))
+						++res[c];
+				}
+			}
+
+			return res;
+		}
+
+		static SmallVector<Candidate, CandidateCount> getCandidatesForFeature(Feature ft, const Result &r)
+		{
+			auto c = getCandidatesForFeature(ft);
+			auto cmp = [&r](Candidate a, Candidate b) {
+				if (r[a] == r[b])
+					return a > b; // prefer candidates with lower enum value
+				return r[a] < r[b];
+			};
+			std::sort(c.begin(), c.end(), cmp);
+
+			return c;
+		}
+
+	private:
+		static SmallVector<Candidate, CandidateCount> getCandidatesForFeature(Feature ft)
+		{
+			switch (ft)
+			{
+			case SubgroupMask:
+				return { NV_shader_thread_group, ARB_shader_ballot };
+			case SubgroupSize:
+				return { NV_shader_thread_group, AMD_gcn_shader, ARB_shader_ballot };
+			case SubgroupInvocationID:
+				return { NV_shader_thread_group, ARB_shader_ballot };
+			case SubgroupID:
+				return { NV_shader_thread_group };
+			case NumSubgroups:
+				return { NV_shader_thread_group };
+			case SubgroupBrodcast_First:
+				return { NV_shader_thread_shuffle, ARB_shader_ballot };
+			case SubgroupBallotFindLSB_MSB:
+				return { NV_shader_thread_group };
+			case SubgroupAll_Any_AllEqual:
+				return { NV_gpu_shader_5, ARB_shader_group_vote, AMD_gcn_shader };
+			case SubgroupElect:
+				return { NV_shader_thread_group };
```

Comment:
we can do our own subgroup elect workaround.

Reply:
@crisspl one more example of something that requires another feature

subgroupElect can be achieved via `LSB` of a ballot, so needs ballot feature.